src/main/java/de/retest/web/RecheckSeleniumAdapter.java

```diff
@@ -53,12 +53,17 @@
 
 	public RecheckSeleniumAdapter( final RecheckOptions options ) {
 		retestIdProvider = options.getRetestIdProvider();
+		screenshotProvider = getScreenshotProvider( options );
+	}
+
+	private ScreenshotProvider getScreenshotProvider( final RecheckOptions options ) {
 		if ( options instanceof RecheckWebOptions ) {
-			final RecheckWebOptions webOptions = (RecheckWebOptions) options;
-			screenshotProvider = webOptions.getScreenshotProvider();
-		} else {
-			screenshotProvider = RecheckWebProperties.getInstance().screenshotProvider();
+			final ScreenshotProvider fromOptions = ((RecheckWebOptions) options).getScreenshotProvider();
+			if ( fromOptions != null ) {
```

Comment:
I quite liked the fact that the `RecheckOptionsBuilder` does the check so that the `RecheckOptions` are properly initialized. That was the whole point around the discussion on who should do the initialization: Builder or Options.

This introduces the same problem as with `reportUploadEnabled` which needs the second evaluation for the properties to determine the actual value. Thus it would be the same with the `RecheckOptions`, if moving the report upload to the builder. 

Thus I would suggest to move the init methods within the options, as they are the first to be called. Or maybe into the `RecheckWebProperties#getInstance`?

As a side note. `RecheckOptions#init` and `RecheckWebOptions#init` does the same. As `ConfigFactory` is a static class, wouldn't one of them suffice?

Reply:
Will resolve this as discussed in Slack.